assets/src/components/form/inputs/form-input.js

```diff
@@ -48,215 +48,206 @@ HTML5_INPUT_TYPES.push( 'iconButton' );
-						htmlClass={ htmlClass }
-						dataSet={ dataSet }
-						helpTextID={ helpTextID }
-						validate={ validations }
-						{ ...attributes }
-					/>
-				);
-				break;
-			case 'radio' :
-				formInput = (
-					<RadioInput
-						name={ name }
-						htmlId={ htmlId }
-						htmlClass={ htmlClass }
-						dataSet={ dataSet }
-						helpTextID={ helpTextID }
-						validate={ validations }
-						{ ...attributes }
-					/>
-				);
-				break;
-			case 'select' :
-				delete attributes.initialValue;
-				formInput = (
-					<SelectInput
-						name={ name }
-						htmlId={ htmlId }
-						htmlClass={ htmlClass }
-						dataSet={ dataSet }
-						helpTextID={ helpTextID }
-						validate={ validations }
-						{ ...attributes }
-					/>
-				);
-				break;
-			case 'textarea' :
-				delete attributes.initialValue;
-				formInput = (
-					<Textarea
-						name={ name }
-						htmlId={ htmlId }
-						htmlClass={ htmlClass }
-						dataSet={ dataSet }
-						helpTextID={ helpTextID }
-						validate={ validations }
-						{ ...attributes }
-					/>
-				);
-				break;
-			case 'date' :
-			case 'month' :
-			case 'datetime-local' :
-				formInput = (
-					<DateTimeInput
-						name={ name }
-						htmlId={ htmlId }
-						htmlClass={ htmlClass }
-						aria-describedby={ helpTextID }
-						{ ...dataSet }
-						{ ...attributes }
-						{ ...validations }
-					/>
-				);
-				break;
-			case 'toggle' :
-				const checked = !! attributes.initialValue;
-				delete attributes.initialValue;
-				delete attributes.inputWidth;
-				formInput = (
-					<ToggleInput
-						name={ name }
-						checked={ checked }
-						htmlId={ htmlId }
-						htmlClass={ htmlClass }
-						aria-describedby={ helpTextID }
-						{ ...dataSet }
-						{ ...attributes }
-						{ ...validations }
-					/>
-				);
-				helpText = '';
-				break;
-			case 'iconButton' :
-				delete attributes.initialValue;
-				delete attributes.inputWidth;
-				formInput = (
-					<IconButtonInput
-						name={ name }
-						htmlId={ htmlId }
-						htmlClass={ htmlClass }
-						aria-describedby={ helpTextID }
-						{ ...attributes }
-					/>
-				);
-				helpText = '';
-				break;
-			default :
-				delete attributes.initialValue;
-				formInput = (
-					<TextInput
-						type={ type }
-						name={ name }
-						htmlId={ htmlId }
-						htmlClass={ htmlClass }
-						dataSet={ dataSet }
-						helpTextID={ helpTextID }
-						validate={ validations }
-						{ ...attributes }
-					/>
-				);
-		}
-		formInput = afterInput ? (
-			<div className={ 'ee-after-input-wrapper' }>
-				{ formInput }
-				<div className={ 'ee-after-input-div' }>{ afterInput }</div>
-			</div>
-		) : formInput;
-		const formListener = isFunction( changeListener ) ? (
-			<OnChange name={ name }>
-				{ changeListener }
-			</OnChange>
-		) : null;
-		return InputLayout ? (
-			<Fragment>
-				<InputLayout
-					inputName={ name }
-					formInput={ formInput }
+const FormInput = ( {
```

Comment:
for some reason i had thought that this one was going to be tricky to convert, 
but it looks like it was really straight forward, good stuff 👍